eclair-core/src/main/scala/fr/acinq/eclair/channel/ChannelFeatures.scala

```diff
@@ -55,18 +62,64 @@ object ChannelFeatures {
 
   def apply(features: Feature*): ChannelFeatures = ChannelFeatures(Set.from(features))
 
-  /** Pick the channel features that should be used based on local and remote feature bits. */
-  def pickChannelFeatures(localFeatures: Features, remoteFeatures: Features): ChannelFeatures = {
+  /** Enrich the channel type with other permanent features that will be applied to the channel. */
+  def apply(channelType: ChannelType, localFeatures: Features, remoteFeatures: Features): ChannelFeatures = {
     // NB: we don't include features that can be safely activated/deactivated without impacting the channel's operation,
     // such as option_dataloss_protect or option_shutdown_anysegwit.
-    val availableFeatures = Set[Feature](
-      StaticRemoteKey,
-      Wumbo,
-      AnchorOutputs,
-      OptionUpfrontShutdownScript
-    ).filter(f => Features.canUseFeature(localFeatures, remoteFeatures, f))
-
-    ChannelFeatures(availableFeatures)
+    val availableFeatures: Seq[Feature] = Seq(Wumbo, OptionUpfrontShutdownScript).filter(f => Features.canUseFeature(localFeatures, remoteFeatures, f))
+    val allFeatures = channelType.features.activated.keys.toSeq ++ availableFeatures
+    ChannelFeatures(allFeatures: _*)
+  }
+
+}
+
+/** A channel type is a specific set of even feature bits that represent persistent channel features as defined in Bolt 2. */
+sealed trait ChannelType {
+  // @formatter:off
+  /** Features representing that channel type. */
+  def features: Features
+  /** True if our main output in the remote commitment is directly sent (without any delay) to one of our wallet addresses. */
+  def paysDirectlyToWallet: Boolean
+  // @formatter:on
+}
+
+object ChannelTypes {
+
+  // @formatter:off
+  case object Standard extends ChannelType {
+    override def features: Features = Features.empty
+    override def paysDirectlyToWallet: Boolean = false
+    override def toString: String = "standard"
+  }
+  case object StaticRemoteKey extends ChannelType {
+    override def features: Features = Features(Features.StaticRemoteKey -> FeatureSupport.Mandatory)
+    override def paysDirectlyToWallet: Boolean = true
+    override def toString: String = "static_remotekey"
+  }
+  case object AnchorOutputs extends ChannelType {
+    override def features: Features = Features(Features.StaticRemoteKey -> FeatureSupport.Mandatory, Features.AnchorOutputs -> FeatureSupport.Mandatory)
+    override def paysDirectlyToWallet: Boolean = false
+    override def toString: String = "anchor_outputs"
+  }
+  // @formatter:on
+
+  // NB: Bolt 2: features must exactly match in order to identify a channel type.
+  def fromFeatures(features: Features): Option[ChannelType] = features match {
+    case f if f == AnchorOutputs.features => Some(AnchorOutputs)
+    case f if f == StaticRemoteKey.features => Some(StaticRemoteKey)
+    case f if f == Standard.features => Some(Standard)
+    case _ => None
+  }
```

Comment:
```suggestion
  def fromFeatures(features: Features): Try[ChannelType] = features match {
    case f if f == AnchorOutputs.features => Success(AnchorOutputs)
    case f if f == StaticRemoteKey.features => Success(StaticRemoteKey)
    case f if f == Standard.features => Success(Standard)
    case _ => Failure(new RuntimeException(s"unknown channel type with features=$features"))
  }
```

Reply:
Done slightly differently in https://github.com/ACINQ/eclair/pull/1867/commits/03bf4c363efa65184c39a2af079fb8976f389b77